### ISSUE TITLE
How to define number of shards and replicas?

### ISSUE THREAD

**Issue by nook24** [crate/crate-dbal#75]
Hi,
I want to use DBAL to manage schema updates.
Unfortunately I don't know how pass shards and replicas.

In addition, when I run this:
```PHP
$connectionParams = [
                'user'        => null,
                'password'    => null,
                'host'        => '127.0.0.1',
                'port'        => 4200,
                'driverClass' => 'Crate\DBAL\Driver\PDOCrate\Driver',
];
$connection = DriverManager::getConnection($connectionParams);
$platform = $connection->getDatabasePlatform();

$SchemaManager = $connection->getSchemaManager();

$schema = $SchemaManager->createSchema();
```

I get:
```
PHP Notice:  Undefined offset: 0 in vendor/crate/crate-dbal/src/Crate/DBAL/Schema/CrateSchemaManager.php on line 62
```
Looks like there is a bug with primary keys.

I'm using version crate-dbal 2.0.0 and CrateDB 4.0.4

**Comment by seut** [comment]
@nook24 Thanks for reporting the missing support for custom CrateDB table options.
I've created a PR to add this.

We'll look into the PK issue next.

**Comment by seut** [comment]
@nook24 We cannot reproduce the PK issue on `$SchemaManager->createSchema();`, we also have a test for validating the functionality using a primary key, see https://github.com/crate/crate-dbal/blob/master/test/Crate/Test/DBAL/Functional/Schema/SchemaManagerTest.php#L117.

Could you please paste us your used table schema?

**Comment by nook24** [comment]
Hi @seut, many thanks for looking into this. This is the schema I used: https://github.com/statusengine/worker/blob/master/lib/crateDB.sql

Maybe this issue is caused by the multi field primary keys

**Comment by seut** [comment]
@nook24 Thanks for the information. We've figured out that the issue is not related to tables *with* primary key constraints but related to tables *without* any.
We're working on a fix, I'll keep you updated.

**Comment by seut** [comment]
@nook24 We've fixed the issue which will be part of the next release.
Thanks for bringing this up.

**Comment by seut** [comment]
@nook24 We've released v2.1.0 including the PK fix and table options support.

**Comment by nook24** [comment]
Many thanks for your fast fix! I will test it :)
